Framework/PythonInterface/plugins/algorithms/ResetNegatives2D.py

```diff
@@ -0,0 +1,65 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid import mtd
+from mantid.api import (AlgorithmFactory, PythonAlgorithm, WorkspaceProperty)
+from mantid.kernel import Direction
+
+
+class ResetNegatives2D(PythonAlgorithm):
+    def category(self):
+        return 'Diffraction\\DataHandling'
+
+    def summary(self):
+        """
+        summary of the algorithm
+        :return:
+        """
+        return "The algorithm used to reset negative values in 2D workspaces."
+
+    def name(self):
+        return "ResetNegatives2D"
+
+    def seeAlso(self):
+        return ["PowderReduceP2D", "Bin2DPowderDiffraction", "SaveP2D"]
+
+    def PyInit(self):
+        self.declareProperty(WorkspaceProperty('Workspace',
+                                               '',
+                                               direction=Direction.Input),
+                             doc='Workspace that should be used.')
+
+    def PyExec(self):
+        data = mtd[self.getPropertyValue('Workspace')]
+        ndp = data.getDimension(1).getNBins()
+        last_dp = -1
+        intMin = 1
+        n = 0
+
+        # Iterate through each Spectrum (dp cuts)
+        for cdp in range(ndp):
```

Comment:
Maybe I'm just not familiar with this terminology, but what is cdp?

Reply:
cdp is short for "cut dp". This refers to the way data is stored. Each spectrum is basically a cut of one specific dp value and we iterate over all spectra at this point.